easy-ui-react/src/FormLayout/Section.tsx

```diff
@@ -0,0 +1,72 @@
+import React, {
+  NamedExoticComponent,
+  ReactNode,
+  isValidElement,
+  useMemo,
+} from "react";
+import { Text } from "../Text";
+import { classNames } from "../utilities/css";
+import { deepFind } from "../utilities/react";
+import { WithinSectionContext } from "./context";
+
+import styles from "./Section.module.scss";
+
+export type SectionProps = {
+  /**
+   * Content of the form section.
+   */
+  children: ReactNode;
+};
+
+export function Section(props: SectionProps) {
+  const { children } = props;
+  const className = classNames(styles.Section);
+
+  // A `fieldset`'s `legend` has to be the first child of a fieldset, so we
+  // search recursively through the Section's children to find the first Title
+  // component and extract the text from it to use in this component
+  const sectionTitle = useMemo(() => {
+    return findSectionTitleInChildren(children);
+  }, [children]);
+
+  return (
+    <WithinSectionContext.Provider value={true}>
+      <fieldset className={className}>
+        {sectionTitle && (
+          <Text as="legend" visuallyHidden>
+            {sectionTitle}
+          </Text>
+        )}
+        {children}
+      </fieldset>
+    </WithinSectionContext.Provider>
+  );
+}
+
+Section.displayName = "FormLayout.Section";
+
+function findSectionTitleInChildren(children: ReactNode) {
+  // look for the first `FormLayout.Title` or `FormLayout.Section` element.
+  // if we encounter a Section before a Title we know the outer Section
+  // doesn't have a title (or it's wildly [and incorrectly] out of order)
```

Comment:
**thought**: not related to whats going on in this function, but this comment got me thinking about error handling for components in easy ui that require some ordering or to be structured in some way. i haven't seen how other libraries deal with it, if it all, but i did some minor handling in the search nav and at the time it made sense for me to do it, but i wonder what the best practices are for these kinds of things.

Reply:
true. it's a good question. i think some libraries handle this situation with "slots", which i think i've seen comes with some error handling, but i'll do some research for future iterations